purview/api/v1.py

```diff
@@ -103,4 +122,110 @@ def get_resource_story(resource, uid):
 
     results['meta']['related_nodes'].update(get_corelated_nodes(results_unordered))
 
+    # Adding the artifact itself if it's story is not available
+    if len(results['data']) == 0:
```

Comment:
Optional: This check could be moved up in the function (with some changes) so that the while loop is not needlessly executed when there is no path.